### PR TITLE
[Examples] Add docker compose example to run multiple containers

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Simple example showing how to use docker compose to launch multiple containers on a SkyPilot cluster.

- [x] `sky launch -c myclus compose_example.yaml`

### CODE REVIEW

**Comment by concretevitamin** [comment]
This is great to have @romilbhardwaj! 

One issue with running it on Azure:
```
» sky launch compose_example.yaml -c dbg --cloud azure --cpus 2+ --down --gpus T4:4
...
(task, pid=26616)  gpu-app1 Pulled
(task, pid=26616)  gpu-app2 Pulled
(task, pid=26616)  Network sky_workdir_default  Creating
(task, pid=26616)  Network sky_workdir_default  Created
(task, pid=26616)  Container sky_workdir-gpu-app2-1  Creating
(task, pid=26616)  Container sky_workdir-gpu-app1-1  Creating
(task, pid=26616)  Container sky_workdir-gpu-app2-1  Created
(task, pid=26616)  Container sky_workdir-gpu-app1-1  Created
(task, pid=26616) Attaching to sky_workdir-gpu-app1-1, sky_workdir-gpu-app2-1
(task, pid=26616) Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: requirement error: unsatisfied condition: cuda>=11.8, please update your driver to a newer version, or use an earlier cuda container: unknown
ERROR: Job 1 failed with return code list: [1]
INFO: Job finished (status: FAILED).
...
```

**Comment by romilbhardwaj** [comment]
Ah good catch - we do need to update our azure image (#2751). For this PR, I've changed the version to 11.5.2 and tested it works on aws, az and gcp.

File: examples/docker/compose/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: '3.8'
+
+services:
+  gpu-app1:
+    image: nvidia/cuda:11.5.2-runtime-ubuntu20.04
+    command: nvidia-smi
```
Comment (concretevitamin) [inline examples/docker/compose/docker-compose.yml:6]: I added `-l 1` to here and L17 for nvidia-smi to loop forever. 

It appears both containers print the same GPU ID. Note the SkyPilot task has 2 GPUs assigned, so GPUs 0 and 1 are available.

Is there any env var (CUDA_VISIBLE_DEVICES?) we can add to this file to show how to distribute the containers to GPUs 0 and 1 respectively? Can even be a comment.

Comment (romilbhardwaj) [reply examples/docker/compose/docker-compose.yml]: Good point - I've changed from `count` to explicit `device_id`. Note that nvidia-docker remaps device ids, so from within `gpu-app2` container the GPU ID visible will be ['0'] (though it maps to physical device 1). Also added this as a comment.

File: examples/docker/compose/docker-compose.yml [hunk 2]
```diff
@@ -3,13 +3,13 @@ version: '3.8'
 services:
   gpu-app1:
     image: nvidia/cuda:11.5.2-runtime-ubuntu20.04
-    command: nvidia-smi
+    command: nvidia-smi # To keep the running in a loop, add -l 1
```
Comment (concretevitamin) [inline examples/docker/compose/docker-compose.yml:6]:
```suggestion
    command: nvidia-smi # To keep running in a loop, add -l 1
```

**Review by concretevitamin — APPROVED** [review]
LGTM!

**Pull request merged**